src/internal/connector/exchange/data_collections.go

```diff
@@ -185,24 +185,27 @@ func DataCollections(
 	}
 
 	for _, scope := range eb.Scopes() {
-		dps := cdps[scope.Category().PathType()]
+		if errs.Failed() {
+			break
+		}
 
 		dcs, err := createCollections(
 			ctx,
 			acct,
 			user,
 			scope,
-			dps,
+			cdps[scope.Category().PathType()],
 			ctrlOpts,
 			su)
 		if err != nil {
-			return nil, support.WrapAndAppend(user, err, errs)
+			errs.Add(err)
+			continue
 		}
 
 		collections = append(collections, dcs...)
 	}
 
-	return collections, errs
+	return collections, errs.Err()
```

Comment:
I think we need to be careful here, it seems like it has the possibility to hide errors if we're running in not fail fast mode

Some questions about this:
* since we now pass in `fault.Errors` is it required to return this?
* is it possible that we need to return the union of `errs.Err()` and `errs.Errs()`? Or different ones depending on if we're in fail fast mode?

Reply:
> since we now pass in fault.Errors is it required to return this?

Yes.  Consider the prior implementation with multi-errors:

```go
var errs error

for := range stuffToDo() {
  if failFast && errs != nil {
    break
  }
  doStuff()
}

return errs.ErrorOrNil()
```

The primary difference between the two implementations is that multierrors would always return an error if one occurred, while fault will only return an error if it is considered a non-recoverable case, which includes checking failFast.  IE: we're now opening  up the ability to continue processing in bestEffort cases.

Returning errs.Err() is necessary because the adding errors via fault provides no guarantees about non-recoverable (or failFast) error cases.  This ensures we don't hide or lose exit cases, while still optionally supporting best effort.